### PR TITLE
Add site_url setting

### PR DESCRIPTION
**Pull request by sevein**
This is connected to #1140.

### CODE REVIEW

File: src/dashboard/src/installer/forms.py [hunk 1]
```diff
@@ -22,8 +22,21 @@
 from django.forms.widgets import TextInput
 from django.utils.translation import ugettext_lazy as _l
 
+from main.models import DashboardSetting
+
+
+site_url_field = forms.CharField(
```
Comment (jraddaoui) [inline src/dashboard/src/installer/forms.py:28]: Could we use `URLField` to improve validation when it's not read-only?

**Review by jraddaoui — APPROVED** [review]
Looks great, just a minor comment.

File: src/dashboard/src/installer/forms.py (hunk 1 above)
Comment (sevein) [reply src/dashboard/src/installer/forms.py:28]: I wanted to but `URLField` won't validate values like `http://archivematica-storage-sevice:8000`. I guess it's looking for fully qualified domain names but if you're on a local network I'm not sure why that should be a requirement.

File: src/dashboard/src/installer/forms.py [hunk 2]
```diff
@@ -22,8 +22,21 @@
...
+    label=_l('Site URL'),
+    help_text=_l('It defines the public address of this service. This is'
+                 ' needed so the service (which is part of the pipeline)'
+                 ' is accessible by Storage Service.'),
```
Comment (jrwdunham) [inline src/dashboard/src/installer/forms.py]: Nitpick: "is accessible by" sounds weird here to me. Maybe "can be accessed by the".

File: src/dashboard/src/installer/forms.py [hunk 3]
```diff
@@ -32,6 +45,10 @@ class Meta:
         model = User
         fields = ['org_name', 'org_identifier', 'username', 'first_name', 'last_name', 'email', 'password1', 'password2']
 
+    def __init__(self, *args, **kwargs):
+        super(OrganizationForm, self).__init__(*args, **kwargs)
```
Comment (jrwdunham) [inline src/dashboard/src/installer/forms.py]: Wrong first argument to `super`; it should be `SuperUserCreationForm` I think.

File: src/dashboard/src/templates/administration/general.html [hunk 4]
```diff
@@ -34,6 +34,10 @@ <h3>{% trans "General configuration" %}</h3>
 
       <form method='POST'>
 
+        {% include "_form.html" with form=general_form %}
+
+        <hr />
+
```
Comment (jrwdunham) [inline src/dashboard/src/templates/administration/general.html]: Seems a bit inconsistent that there's a horizontal rule here and then there are `<h4>`s for "Storage Service options" and "Elasticsearch indexing", etc. but no `<h4>` header for this section. Super nitpicky yet vague formatting comment, so feel free to ignore.

File: src/dashboard/src/components/administration/views.py [hunk 5]
```diff
@@ -450,15 +451,17 @@ def general(request):
...
                                        prefix='storage', initial=initial_data)
     checksum_form = ChecksumSettingsForm(request.POST or None,
                                          prefix='checksum algorithm',
                                          initial=initial_data)
 
-    if storage_form.is_valid() and checksum_form.is_valid():
-        storage_form.save()
-        checksum_form.save()
+    forms = (general_form, storage_form, checksum_form)
+    if map(lambda form: form.is_valid(), forms):
```
Comment (jrwdunham) [inline src/dashboard/src/components/administration/views.py]: I'm not sure this condition is correct. No matter how many of the forms are valid or invalid, the return value of `map` is still going to be something that is `True` when `bool` is applied to it.

**Review by jrwdunham — CHANGES_REQUESTED** [review]
I think there's one or two problems with this PR and I have some other minor comments.

File: src/dashboard/src/components/administration/views.py (hunk 5 above)
Comment (sevein) [reply src/dashboard/src/components/administration/views.py]: Ops I meant to use `all(map(...))`

**Review by jrwdunham — APPROVED** [review]

**Pull request merged**